pennylane/templates/state_preparations/mottonen.py

```diff
@@ -132,19 +156,19 @@ def _apply_uniform_rotation_dagger(gate, alpha, control_wires, target_wire):
         return op_list
 
     code = gray_code(gray_code_rank)
-    num_selections = len(code)
-
-    control_indices = [
-        int(np.log2(int(code[i], 2) ^ int(code[(i + 1) % num_selections], 2)))
-        for i in range(num_selections)
-    ]
-
+    control_indices = np.log2(code ^ np.roll(code, -1)).astype(int)
+
+    # For abstract or differentiated theta we will never skip a rotation. Likewise if there is at
+    # least one non-zero angle (per batch if batched) for all rotations.
+    skip_none = qml.math.is_abstract(theta) or qml.math.requires_grad(theta)
+    if not skip_none:
```

Comment:
I understand that you skip for abstract, but I'm wondering if this is going to be an issue in the future. Using jit we are going to start to have less efficient implementations right?

Reply:
No start.  We've been that for years.